index.js

```diff
@@ -8,6 +8,12 @@ var util = require("util");
 var exec = require("child_process").exec;
 
 function PDFImage(pdfFilePath, options) {
+  // validating the file path for invalid characters to prevent remote code execution
+  if (/;|&|`|\$|\(|\)|\|\||\||!|>|<|\?|\${/g.test(pdfFilePath)) {
```

Comment:
The regexp isn't easily readable and so maintainable and especially two parts of the expression.
The first part is `|\|\||\|` and the second is `|\${`.